code/modules/projectiles/guns/energy/blaster.dm

```diff
@@ -116,3 +116,63 @@
 		toggle_scope(2.0, usr)
 	else
 		to_chat(usr, "<span class='warning'>You can't look through the scope without stabilizing the rifle!</span>")
+
+
+/obj/item/gun/energy/disruptorpistol
+	name = "disruptor pistol"
+	desc_fluff = "Developed and produced by NanoTrasen for its internal security division, the NT DP-7 is a state of the art blaster pistol capable of firing reduced-power bolts which disrupt the central nervous system, inducing a stunning effect on the victim. It is also capable of firing full-power blaster bolts."
+	desc = "A Nanotrasen designed blaster pistol carbine with two settings: stun and lethal."
+	icon = 'icons/obj/guns/disruptorpistol/disruptorpistols.dmi'
+	icon_state = "disruptorpistol"
+	item_state = "disruptorpistol"
+	fire_sound = 'sound/weapons/gunshot/bolter.ogg'
+	slot_flags = SLOT_BELT|SLOT_HOLSTER
+	w_class = ITEMSIZE_SMALL
+	force = 5
+	origin_tech = list(TECH_COMBAT = 3, TECH_MAGNET = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	projectile_type = /obj/item/projectile/energy/disruptorstun
+	secondary_projectile_type = /obj/item/projectile/energy/blaster
+	max_shots = 12 //12 shots stun, 8 shots lethal.
+	charge_cost = 50
+	has_item_ratio = FALSE
+	modifystate = "disruptorpistolstun"
+	sel_mode = 1
+	var/selectframecheck = FALSE
+	firemodes = list(
+		list(mode_name="stun", projectile_type=/obj/item/projectile/energy/disruptorstun, modifystate="disruptorpistolstun", charge_cost = 50, fire_sound = 'sound/weapons/gunshot/bolter.ogg'),
+		list(mode_name="lethal", projectile_type=/obj/item/projectile/energy/blaster, modifystate="disruptorpistolkill", recoil = 1, charge_cost = 75, fire_sound = 'sound/weapons/gunshot/bolter.ogg')
+		)
+
+/obj/item/gun/energy/disruptorpistol/security
+	pin = /obj/item/device/firing_pin/wireless
+
+/obj/item/gun/energy/disruptorpistol/verb/select_frame() // Let's people choose the aesthetic model of their gun. No stat changes.
+	set name = "Select Model"
+	set category = "Object"
+	set desc = "Click to select the model of your gun."
+
+	var/mob/M = usr
+	var/user_reply
+
+	if(!M.mind)
+		return FALSE
+
+	user_reply = input("Select your frame.") in list("sub-compact","service","magnum")
+	if(!use_check_and_message(usr))
+		if(user_reply == "sub-compact")
+			icon = 'icons/obj/guns/disruptorpistol/disruptorpistolc.dmi'
+		if(user_reply == "service")
+			icon = 'icons/obj/guns/disruptorpistol/disruptorpistols.dmi'
+		if(user_reply == "magnum")
+			icon = 'icons/obj/guns/disruptorpistol/disruptorpistolm.dmi'
```

Comment:
this should be a radial menu so people can actually see what the icon will be

Reply:
Having the icon be tweakable to such a degree just feels weird in the first place, when they'll have the same stats no matter what. Unless I'm missing something.